parquet-column/src/main/java/org/apache/parquet/column/Encoding.java

```diff
@@ -189,13 +189,29 @@ public ValuesReader getValuesReader(ColumnDescriptor descriptor,
    * Suffixes are stored as delta length byte arrays.
    */
   DELTA_BYTE_ARRAY {
+    private DeltaByteArrayReader previous;
```

Comment:
This is a good fix for the bug, but now this class and its getter are stateful, which is why I am a little worried about putting it directly into parquet instead of a branch or repair tool or something.